File: common-rule-engine/src/test/files/checks/SwitchWithTooManyCaseCheck/SwitchWithTooManyCaseCheckMax5.go
package main
func foo(tag int) {
   switch tag { // Noncompliant {{Reduce the number of switch cases from 31 to at most 5.}}
// ^^^^^^
	default:
		bar()
	case 0, 1, 2, 3:
		bar()
	case 4, 5, 6, 7:
		bar()
  case 4, 5, 6, 7:
    bar()
  case 4, 5, 6, 7:
    bar()
  case 0, 1, 2, 3:
    bar()
  case 4, 5, 6, 7:
    bar()
  case 4, 5, 6, 7:
    bar()
  case 4, 5, 6, 7:
    bar()
  case 0, 1, 2, 3:
    bar()
  case 4, 5, 6, 7:
    bar()
  case 0, 1, 2, 3:
    bar()
  case 4, 5, 6, 7:
    bar()
  case 4, 5, 6, 7:
    bar()
  case 4, 5, 6, 7:
    bar()
  case 0, 1, 2, 3:
    bar()
  case 4, 5, 6, 7:
    bar()
  case 4, 5, 6, 7:
    bar()
  case 4, 5, 6, 7:
    bar()
  case 0, 1, 2, 3:
    bar()
  case 4, 5, 6, 7:
    bar()
  case 0, 1, 2, 3:
    bar()
  case 4, 5, 6, 7:
    bar()
  case 4, 5, 6, 7:
    bar()
  case 4, 5, 6, 7:
    bar()
  case 0, 1, 2, 3:
    bar()
  case 4, 5, 6, 7:
    bar()
  case 4, 5, 6, 7:
    bar()
  case 4, 5, 6, 7:
    bar()
  case 0, 1, 2, 3:
    bar()
  case 4, 5, 6, 7:
    bar()
	}

}

func foo2(tag int) {
  switch tag { // Noncompliant
//^^^^^^
  default:
//^^^^^^^< {{+1}}
		bar()
  case 0, 1, 2, 3:
//^^^^< {{+1}}
	bar()
  case 4, 5, 6, 7:
//^^^^< {{+1}}
		bar()
  case 4, 5, 6, 7:
//^^^^< {{+1}}
    bar()
  case 4, 5, 6, 7:
//^^^^< {{+1}}
    bar()
  case 0, 1, 2, 3:
//^^^^< {{+1}}
    bar()
  case 4, 5, 6, 7:
//^^^^< {{+1}}
    bar()
  }
}
func bar(s ... string) {
}

File: common-rule-engine/src/test/files/checks/SwitchWithTooManyCaseCheck/SwitchWithTooManyCaseCheckMax5.java
class A{
  void foo() {
    switch (1) { // Noncompliant {{Reduce the number of non-empty switch cases from 38 to at most 5.}}
//	^^^^^^
      case 1:
        System.out.println("");
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      System.out.println("");
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
        break;
      case 1:
      case 1:
      case 1:
      case 1:
        break;
      System.out.println("");
      default:
        System.out.println("");
    }

    switch (1) { // Noncompliant
//  ^^^^^^
      case 1:
      case 1:
      case 1:
        System.out.println("");
      case 1:
      case 1:
      case 1:
        System.out.println("");
    }
  }
}
